lib/queue.js

```diff
@@ -830,16 +830,40 @@ class Queue extends Emitter {
       args[i] = arguments[i];
     }
 
-    return this._commandable().then((client) => {
-      // Get the sha for the script after we're ready to avoid a race condition
-      // with the lua script loader.
-      args[0] = lua.shas[name];
-
-      const promise = helpers.deferred();
-      args.push(promise.defer());
-      client.evalsha.apply(client, args);
-      return promise;
-    });
+    return this._commandable()
+      .then((client) => {
+        // Get the sha for the script after we're ready to avoid a race
+        // condition with the lua script loader.
+        args[0] = lua.shas[name];
+
+        const promise = helpers.deferred();
+        args.push(promise.defer());
+        client.evalsha.apply(client, args);
+        return promise;
+      })
+      .catch((err) => {
+        // In cases where the Redis server fails over, the script cache is
+        // cleared (even when the cache itself is still intact.) We detect this
+        // condition and reload the scripts.
+        if (
+          err.name === 'ReplyError' &&
+          err.message === 'NOSCRIPT No matching script. Please use EVAL.'
+        ) {
+          return lua.buildCache(this.client).then((client) => {
+            // Try again.
+            // Get the sha for the script after we're ready to avoid a race
+            // condition with the lua script loader.
+            args[0] = lua.shas[name];
```

Comment:
`args` will already have been modified in the preceding `then`, we just need to replace the last element with a fresh `helpers.deferred().defer()`

Reply:
I believe I've hit that note in the latest commit; lemme know if that's wrong.